### PR TITLE
Vasilis/forest dml docs

### PR DESCRIPTION
**Pull request by vasilismsr**
Revamped all our user guides.

### CODE REVIEW

File: doc/spec/estimation/dml.rst [hunk 1]
```diff
@@ -1,3 +1,5 @@
+.. _druserguide:
```
[the remark this answers is not in the document]
Comment (vasilismsr) [reply doc/spec/estimation/dml.rst]: What do you mean?

File: doc/reference.rst [hunk 2]
```diff
@@ -14,7 +14,6 @@ Public Module Reference
     econml.inference
     econml.ortho_forest
     econml.metalearners
-    econml.drlearner
```
[the remark this answers is not in the document]
Comment (vasilismsr) [reply doc/reference.rst]: This is a repeat. See a few lines above.

File: doc/spec/estimation/dml.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (vasilismsr) [reply doc/spec/estimation/dml.rst]: oh got it

File: doc/spec/estimation/dr.rst [hunk 3]
```diff
@@ -0,0 +1,488 @@
+.. _dmluserguide:
```
[the remark this answers is not in the document]
Comment (vasilismsr) [reply doc/spec/estimation/dr.rst]: done

File: doc/reference.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (kbattocchi) [reply doc/reference.rst]: Ah, probably a bad merge previously.

**Pull request merged**